### DISCUSSION TITLE
Is there any way reject WebSocket connection

### DISCUSSION THREAD

**Discussion by cwhan-yonsei** [tokio-rs/axum#1627] [Q&A] [answered]
Sometimes a WebSocket upgrade from a client needs to be rejected. 
(e.g. upgrade request to wrong URI or client authentication)
However, it seems there is no way to reject the WebSocket upgrade request but to close the connection immediately. 
- https://docs.rs/axum/latest/axum/extract/struct.WebSocketUpgrade.html#
Is there any way to reject the upgrade?

**Comment by davidpdrsn** [comment, marked as answer]
Just don't call `on_upgrade` and return a different response.
